CMakeLists.txt

```diff
@@ -67,86 +48,101 @@ set( WIN32_BOOTSTRAP_SOURCES
 
 # ==============================================================================
 
-function(_COMMON_PROJECT_OPTIONS)
-	include_directories(${INCLUDE_DIR} ${LIB_INCLUDE_DIR})
-	link_directories(${LIB_DIR}/boost/lib/)
-
-	add_compile_options(-std:c++latest)
+function(_COMMON_PROJECT_OPTIONS TARGET_NAME)
+	target_include_directories(${TARGET_NAME}
+		PRIVATE
+			sources/inc
+	)
+	target_link_libraries(${TARGET_NAME}
+		PRIVATE
+			Boost::boost
+			Boost::filesystem
```

Comment:
Whether it is through `FindBoost` or automatic dependency, you should now used proper CMake targets for this